packages/instant/src/components/amount_input.tsx

```diff
@@ -0,0 +1,42 @@
+import { BigNumber } from '@0xproject/utils';
+import * as _ from 'lodash';
+import * as React from 'react';
+
+import { ColorOption } from '../style/theme';
+
+import { Container, Input } from './ui';
+
+export interface AmountInputProps {
+    fontColor?: ColorOption;
+    fontSize?: string;
+    value?: BigNumber;
+    onChange?: (value?: BigNumber) => void;
+}
+
+export class AmountInput extends React.Component<AmountInputProps> {
+    public render(): React.ReactNode {
+        const { fontColor, fontSize, value } = this.props;
+        return (
+            <Container borderBottom="1px solid rgba(255,255,255,0.3)" display="inline-block">
+                <Input
+                    fontColor={fontColor}
+                    fontSize={fontSize}
+                    onChange={this._handleChange}
+                    value={value ? value.toString() : ''}
+                    placeholder="0.00"
+                    width="2em"
+                />
+            </Container>
+        );
+    }
+    private readonly _handleChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+        const value = event.target.value;
+        let bigNumberValue;
+        if (!_.isEmpty(value)) {
+            bigNumberValue = new BigNumber(event.target.value);
+        }
+        if (!_.isUndefined(this.props.onChange)) {
```

Comment:
Based on the code here, I think if you change the AmountInput's value to be "empty", i.e. changing from "300" to "", we will not trigger the `this.props.onChange` event.   Seems OK, but just wanted to confirm that this is desired behavior.  

Something that's a little funky, is that `_.isEmpty` returns `true` for `''` but `false` for  `'       '`

Reply:
Are we not? Seems like onChange is called for as long as it's defined. It will just be called with undefined as the value since there is no `undefined` or `empty` value for `BigNumber`

Reply:
Whoops, read the code incorrectly -- you are correct.

Per convo, did discover an issue while testing this though: if you type in an invalid number (i.e. letters or a bunch of blank spaces), a JS error is thrown